staging/src/k8s.io/client-go/transport/config.go

```diff
@@ -53,7 +54,7 @@ type Config struct {
 	WrapTransport func(rt http.RoundTripper) http.RoundTripper
 
 	// Dial specifies the dial function for creating unencrypted TCP connections.
-	Dial func(network, addr string) (net.Conn, error)
+	Dial func(ctx context.Context, network, address string) (net.Conn, error)
```

Comment:
intentionally left as Dial and not DialContext?

Reply:
Yes. It's our interface and I thought if we're making a breaking change anyway, lets not make the name ugly.

Reply:
Ok, fine with me

Reply:
could we make sure this breaking change is captured in the release note somehow? probably start with documenting it in this PR's release note section.

Reply:
Added a release note to the PR description.